cli/formatters.py

```diff
@@ -109,7 +110,10 @@ def fmt_list(self, xs: list) -> str:
                 restval="<null>"
             )
 
-            w.writeheader()
+            if not self.wroteHeader:
```

Comment:
This approach is wired. The header should be written once and if you need to test for it, it means that the headers shouldn't be written in this function. 
It's not obvious where is the right place for it, so for now, we may leave it as is but this is worthy of a `TODO`.